loopin.go

```diff
@@ -336,34 +334,53 @@ func (s *loopInSwap) executeSwap(globalCtx context.Context) error {
 
 // waitForHtlcConf watches the chain until the htlc confirms.
 func (s *loopInSwap) waitForHtlcConf(globalCtx context.Context) (
-	*chainntnfs.TxConfirmation, error) {
+	*chainntnfs.TxConfirmation, *swap.Htlc, error) {
 
 	ctx, cancel := context.WithCancel(globalCtx)
 	defer cancel()
-	confChan, confErr, err := s.lnd.ChainNotifier.RegisterConfirmationsNtfn(
-		ctx, nil, s.htlc.PkScript, 1, s.InitiationHeight,
+
+	notifier := s.lnd.ChainNotifier
+
+	confChanP2WSH, confErrP2WSH, err := notifier.RegisterConfirmationsNtfn(
+		ctx, nil, s.htlcP2WSH.PkScript, 1, s.InitiationHeight,
 	)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	confChanNP2WSH, confErrNP2WSH, err := notifier.RegisterConfirmationsNtfn(
+		ctx, nil, s.htlcNP2WSH.PkScript, 1, s.InitiationHeight,
+	)
+	if err != nil {
+		return nil, nil, err
 	}
+
 	for {
 		select {
 
 		// Htlc confirmed.
-		case conf := <-confChan:
-			return conf, nil
+		case conf := <-confChanP2WSH:
+			return conf, s.htlcP2WSH, nil
+
+		// Htlc confirmed.
+		case conf := <-confChanNP2WSH:
+			return conf, s.htlcNP2WSH, nil
```

Comment:
Don't you want to write the htlc type into a field of `swap`? Haven't thought about what is preferable, but maybe it is good for consistency with the server impl

Reply:
How would that work? We need both htlcs for (external) loop-in.

Reply:
I mean once the htlc has confirmed. At that point we only need to deal with one. You return the htlc at this line and then pass it around to several calls.

Reply:
As we need to pass around both htlcs in the `SwapInfo` for monitoring it'd be a larger refactor to do this cleanly which imho is not worth it as it'd just create a lot of duplicated code lines. So instead I simply added an `htlc` member to the `swapKit` struct which points to confirmed htlc when looping in and the p2wsh htlc when looping out.